admin/load.php

```diff
@@ -540,21 +543,29 @@ function perflab_enqueue_modules_page_scripts() {
  */
 function perflab_activate_plugin() {
 	// Do not proceed if plugin query arg is not present.
-	if ( empty( $_GET['plugin'] ) ) {
+	if ( empty( $_GET['plugin_slug'] ) ) {
 		return;
 	}
 
-	// The plugin being activated.
-	$plugin = sanitize_text_field( wp_unslash( $_GET['plugin'] ) );
+	$plugin_slug = sanitize_text_field( wp_unslash( $_GET['plugin_slug'] ) );
 
-	check_admin_referer( "perflab_activate_plugin_{$plugin}" );
+	check_admin_referer( "perflab_activate_plugin_{$plugin_slug}" );
 
-	if ( ! current_user_can( 'activate_plugin', $plugin ) ) {
+	$plugins = get_plugins( '/' . $plugin_slug );
+
+	if ( empty( $plugins ) ) {
+		return;
+	}
+
+	$keys        = array_keys( $plugins );
+	$plugin_file = $plugin_slug . '/' . $keys[0];
```

Comment:
How is it that the first key is the one we know needs to be used?

Reply:
See above, it'll always be just the one plugin (or none if it's not found for some reason).